agent/src/beerocks/slave/backhaul_manager/backhaul_manager_thread.cpp

```diff
@@ -1328,9 +1328,46 @@ bool backhaul_manager::backhaul_fsm_wireless(bool &skip_select)
 
             // Create a HAL instance if doesn't exists
             if (!soc->sta_wlan_hal) {
+
+                // read slave config file
+                std::string slave_config_file_path = "./" + std::string(BEEROCKS_AGENT) +
+                                                     ".conf"; //search first in current directory
+                beerocks::config_file::sConfigSlave beerocks_slave_conf;
+                if (!beerocks::config_file::read_slave_config_file(slave_config_file_path,
+                                                                   beerocks_slave_conf)) {
+                    slave_config_file_path = BEEROCKS_CONF_PATH + std::string(BEEROCKS_AGENT) +
+                                             ".conf"; // if not found, search in beerocks path
+                    if (!beerocks::config_file::read_slave_config_file(slave_config_file_path,
+                                                                       beerocks_slave_conf)) {
+                        std::cout << "config file '" << slave_config_file_path << "' args error."
+                                  << std::endl;
+                        return 1;
+                    }
+                }
```

Comment:
I don't understand the reason of read config file here.

1) `backhaul_manager` constructor receives as argument the `config_file::sConfigSlave` object which already contain data from beerocks agent config file
https://github.com/prplfoundation/prplMesh/blob/c70a5977e8e57a1a8ce6c9b24152381161b186c3/agent/src/beerocks/slave/backhaul_manager/backhaul_manager_thread.cpp#L228

2) You may extend `struct SBackhaulConfig` with array of wpa supplicant ctrl paths and copy them from `sConfigSlave` object

Reply:
I created a separate [issue](https://jira.prplfoundation.org/browse/PPM-232) for fixing commentaries